README.md

```diff
@@ -89,112 +130,185 @@ var requestObject = new RequestObject({
+  context: RequestContext { connectionId: null, protocol: null, token: null, user: null } }
 ```
 
-## `Models.ResponseObject`
+## `models.RequestContext`
 
-This constructor creates a standardized Kuzzle Response object from a `RequestObject` and response data.
+This constructor is used to create a connection context used by `Request` 
 
-#### `ResponseObject(requestObject, responseData)`
+### `new RequestContext([options])`
 
-##### Arguments
+**Arguments**
 
 | Name | Type | Description                      |
 |------|------|----------------------------------|
-|`requestObject`|`RequestObject object`| The request object that generated the response data |
-|`responseData`|`object`| Plain-old JSON object containing the request's results |
+| `options` | `object` | Optional initialization parameters |
 
-##### ResponseObject attributes
+`options` may contain the following attributes:
 
 | Name | Type | Description                      |
 |------|------|----------------------------------|
+| `connectionId` | `string` | Client's connection unique ID |
+| `protocol` | `string` | Network protocol name |
+| `token` | `object` | Kuzzle internal authorization token object |
+| `user` | `object` | Kuzzle internal user info object |
+
+### Attributes
+
+| Name | Type | Description                      |
+|------|------|----------------------------------|
+| `connectionId` | `string` | Client's connection unique ID |
+| `protocol` | `string` | Network protocol name |
+| `token` | `object` | Kuzzle internal authorization token object |
+| `user` | `object` | Kuzzle internal user info object |
+
+## `models.RequestInput`
+
+Contains the request's input data 
+
+### `new RequestInput(data)`
+
+**Arguments**
+
+| Name | Type | Description                      |
+|------|------|----------------------------------|
+|`data`|`object`| Standardized API request (see Websocket requests for instance) |
+
+`data` may contain some or all of the following attributes:
+
+| Name | Type | Description                      |
+|------|------|----------------------------------|
+|`_id`|`string`| Document unique identifier |
 |`action`|`string`| Kuzzle action to perform |
+|`body`|`object`| Contains request specific data (document content, search queries, ...) |
 |`collection` |`string` | Data collection |
 |`controller` |`string`| Kuzzle controller handling the action to perform |
-|`data._id`|`string`| Document unique identifier |
-|`data.body`|`object`| Contains response data (document content, search results, ...) |
-|`error`|`Error object`| In case of error, contains the error object |
 |`index` |`string`| Data index |
-|`metadata`|`object`| Contains request specific metadata |
-|`status`|`integer`| Request status, using [HTTP status codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes) |
-|`timestamp`|`integer`| Request timestamp |
+|`metadata`|`object`| Client's request specific metadata |
+|`token`|`string`| JWT Authentication token |
```

Comment:
OK, forget it, it's not the same one : 
* in `Request.input` : the JWT token string coming from request headers
* in `Request.context`: the token object (decrypted from the input's jwt) used internally within Kuzzle kernel.

It's quite disturbing, while reading the code, to see 2 different representations of `token` attributes...

Maybe we could rename Request input's `token` attribute to `jwt` ?